### PR TITLE
Sprinkle some async/await all over the place

### PR DESCRIPTION
**Pull request by jbuck**

### CODE REVIEW

**Comment by jbuck** [comment]
@philbooth I turned marketing.js into a library that each command (local filesystem or sqs/s3) can use separately. The last bit I'm trying to solve is when to resolve the Promise in [marketing.js](https://github.com/mozilla/fxa-amplitude-send/pull/69/files#diff-65b2cde4fe6b778426eeb027497fa9dfL96)... I think the problem is that the csv-parser keeps feeding new rows in, even if the function is suspended waiting for the promise to return. Not too sure the best way to solve this, it sounds like we'd need some sort of queue within the Promise that calls resolve/reject but I haven't really thought this through

**Comment by philbooth** [comment]
> I think the problem is that the csv-parser keeps feeding new rows in, even if the function is suspended waiting for the promise to return.

Yep, although that's a feature of node streams rather than `csv-parser`. You need to call [`pause`](https://nodejs.org/api/stream.html#stream_readable_pause) if we want to stop the data arriving.

I'm not sure I quite understand exactly what issue you're trying to fix at the moment though. Can jump on vidyo and work through it together if you like?

There's a couple of other (small) tweaks that I'd suggest/make inside the `processStream` function, but I don't want to do anything until I'm sure I understand everything.

**Comment by jbuck** [comment]
Okay, using async made this pretty easy. @philbooth r?

File: marketing.js [hunk 1]
```diff
@@ -3,113 +3,65 @@
...
+      console.log(body)
     }
 
-    return sendBatch(localBatch)
-      .then(() => eventCount += localBatch.length)
-      .catch(e => {
-        error = e
-        throw error
-      })
+    return Promise.resolve(eventCount)
```
Comment (philbooth) [inline marketing.js]: This `Promise.resolve` is unnecessary now it's `async`.

File: marketing.js [hunk 2]
```diff
@@ -3,113 +3,65 @@
...
       })
   })
 
-  function send (localBatch) {
-    if (error) {
-      return Promise.reject(error)
+  async function send (localBatch) {
+    const body = await sendBatch(localBatch)
+
+    if (body == "success") {
```
Comment (philbooth) [inline marketing.js]: Personally this feels like coupling too tightly to an implementation detail of the Amplitude API. If they change the response body at any point we're going to have to fix the script and re-send a bunch of failed events. I thought the previous error-handling strategy was more robust.

But I'm not the one maintaining it, so no biggie if this is preferable. Just wanted to call it out.

Comment (philbooth) [reply marketing.js]: > fix the script and re-send a bunch of failed events

Well, not the re-send part, the events will have been sent successfully of course.

**Pull request merged**